File: app/src/main/java/cn/edu/sdu/online/isdu/bean/Confession.java
package cn.edu.sdu.online.isdu.bean;

import java.util.Objects;

import cn.edu.sdu.online.isdu.interfaces.Collectible;
import cn.edu.sdu.online.isdu.util.history.HistoryRecord;

/**
 ****************************************************
 * @author zsj
 * Last Modifier: ZSJ
 * Last Modify Time: 2018/5/26
 *
 * 论坛帖子的Java Bean
 ****************************************************
 */

public class Confession extends AbstractPost {

    private int confessionId; // 帖子ID
    private int type; // 论坛内容类型
    private String titleFlag;
    //    private String title;
//    private String uid;
    private int commentsNumbers;
    //    private Long time;
//    private String content; // 具体内容
    private int likeNumber;
    private int collectNumber;
    private double value;
    private String tag;

    public Confession() {}

    public Confession(int type, String content) {
        this.type = type;
//        this.content = content;
        this.mContent = content;
    }

    public String getTag() {
        return tag;
    }

    public void setTag(String tag) {
        this.tag = tag;
    }

    public int getConfessionId() {
        return confessionId;
    }

    public void setConfessionId(int confessionId) {
        this.confessionId = confessionId;
    }

    public int getLikeNumber() {
        return likeNumber;
    }

    public void setLikeNumber(int likeNumber) {
        this.likeNumber = likeNumber;
    }

    public int getCollectNumber() {
        return collectNumber;
    }

    public void setCollectNumber(int collectNumber) {
        this.collectNumber = collectNumber;
    }

    public String getTitleFlag() {
        return titleFlag;
    }

    public void setTitleFlag(String titleFlag) {
        this.titleFlag = titleFlag;
    }

    public String getTitle() {
        return mTitle;
    }

    public void setTitle(String title) {
        this.mTitle = title;
    }

    public String getUid() {
        return mUserId + "";
    }

    public void setUid(String uid) {
        this.mUserId = Integer.parseInt(uid);
    }

    public int getCommentsNumbers() {
        return commentsNumbers;
    }

    public void setCommentsNumbers(int commentsNumbers) {
        this.commentsNumbers = commentsNumbers;
    }

    public Long getTime() {
        return mTime;
    }

    public void setTime(Long time) {
        this.mTime = time;
    }

    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }

    public String getContent() {
        return mContent;
    }

    public void setContent(String content) {
        this.mContent = content;
    }

    public double getValue() {
        return value;
    }

    public void setValue(double value) {
        this.value = value;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Confession confession = (Confession) o;
        return confessionId == confession.confessionId;
    }

    @Override
    public int hashCode() {
        return Objects.hash(confessionId);
    }

    @Override
    public void onCollect() {

    }

    @Override
    public void onScan() {
        HistoryRecord.INSTANCE.newHistory(this);
    }
}

File: app/src/main/java/cn/edu/sdu/online/isdu/bean/Confession.kt
package cn.edu.sdu.online.isdu.bean

import android.os.Parcel
import android.os.Parcelable
import java.util.Objects

import cn.edu.sdu.online.isdu.interfaces.Collectible
import cn.edu.sdu.online.isdu.util.history.HistoryRecord

/***********************************************************************
 * @author Grapes
 * @date 2018-12-13
 * @Description 表白帖子信息
 ***********************************************************************/

class Confession() : AbstractPost(), Parcelable {

    var confessionId: Int = 0 // tid
    var commentCount: Int = 0   // the number of comments
    var postType: Int = 0

    var likeCount: Int = 0      // the number of likes
    var value: Double = 0.0
    var tag: String = ""     // the tag of it, maybe it's useless

    var title: String
        get() = mTitle
        set(title) {
            this.mTitle = title
        }

    var uid: String
        get() = mUserId.toString()
        set(uid) {
            this.mUserId = Integer.parseInt(uid)
        }

    var postContent: String
        get() = mContent
        set(value) {
            mContent = value
        }

    var time: Long
        get() = mTime
        set(time) {
            this.mTime = time
        }

    constructor(parcel: Parcel) : this() {
        confessionId = parcel.readInt()
        commentCount = parcel.readInt()
        postType = parcel.readInt()
        likeCount = parcel.readInt()
        value = parcel.readDouble()
        tag = parcel.readString()
    }

    // temporarily unavailable
    /*
    constructor(type: Int, content: String) : this() {
        postType = type
        mContent = content
    }
    */

    override fun getType(): Int {
        return postType
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val confession = o as Confession?
        return confessionId == confession!!.confessionId
    }

    override fun hashCode(): Int {
        return Objects.hash(confessionId)
    }

    override fun onCollect() {

    }

    override fun onScan() {
        HistoryRecord.newHistory(this)
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(confessionId)
        parcel.writeInt(commentCount)
        parcel.writeInt(postType)
        parcel.writeInt(likeCount)
        parcel.writeDouble(value)
        parcel.writeString(tag)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Confession> {
        override fun createFromParcel(parcel: Parcel): Confession {
            return Confession(parcel)
        }

        override fun newArray(size: Int): Array<Confession?> {
            return arrayOfNulls(size)
        }
    }
}
